app/stac_api/validators.py

```diff
@@ -18,63 +20,108 @@
-    ('application/x.netcdf+zip', 'Zipped NetCDF', ['.zip']),
-    ('application/xml', 'XML', ['.xml']),
-    ('application/x.xml+zip', 'Zipped XML', ['.zip']),
-    ('application/vnd.mapbox-vector-tile', 'mbtiles', ['.mbtiles']),
-    ('text/plain', 'Text', ['.txt']),
-    ('text/x.plain+zip', 'Zipped text', ['.zip']),
-    ('application/x.dwg+zip', 'Zipped DWG', ['.zip']),
-    ('application/zip', 'Generic Zip File', ['.zip']),
-    ('image/tiff', 'TIFF', ['.tiff', '.tif']),
-    ('image/jpeg', 'JPEG', ['.jpeg', '.jpg']),
-    ('image/png', 'PNG', ['.png']),
-    ('application/vnd.sqlite3', 'sqlite', ['.sqlite']),
+    MediaType('application/pdf', 'PDF', ['.pdf']),
+    MediaType('application/x.pdf+zip', 'Zipped PDF', ['.zip']),
+    MediaType('application/json', 'JSON', ['.json']),
+    MediaType('application/x.json+zip', 'Zipped JSON', ['.zip']),
+    MediaType('application/x-netcdf', 'NetCDF', ['.nc']),
+    MediaType('application/x.netcdf+zip', 'Zipped NetCDF', ['.zip']),
+    MediaType('application/xml', 'XML', ['.xml']),
+    MediaType('application/x.xml+zip', 'Zipped XML', ['.zip']),
+    MediaType('application/vnd.mapbox-vector-tile', 'mbtiles', ['.mbtiles']),
+    MediaType('text/plain', 'Text', ['.txt']),
+    MediaType('text/x.plain+zip', 'Zipped text', ['.zip']),
+    MediaType('application/x.dwg+zip', 'Zipped DWG', ['.zip']),
+    MediaType('application/zip', 'Generic Zip File', ['.zip']),
+    MediaType('image/tiff', 'TIFF', ['.tiff', '.tif']),
+    MediaType('image/jpeg', 'JPEG', ['.jpeg', '.jpg']),
+    MediaType('image/png', 'PNG', ['.png']),
+    MediaType('application/vnd.sqlite3', 'sqlite', ['.sqlite']),
 ]
-MEDIA_TYPES_MIMES = [x[0] for x in MEDIA_TYPES]
-MEDIA_TYPES_EXTENSIONS = [ext for media_type in MEDIA_TYPES for ext in media_type[2]]
-MEDIA_TYPES_BY_TYPE = {media[0]: media for media in MEDIA_TYPES}
+
+quoted_string_pattern = re.compile('^"[^"]+"$')
+
+
+def get_mime_comparable(mime_string):
+    '''
+    Generates a tuple out of a MIME string that will be equal to all equivalent MIME strings.
+
+    The rules for equivalence are:
+    - Types, subtypes and parameter names are case insensitive.
+    - Parameter values may be case sensitive (this function interprets this statement as "always
+      case sensitive")
+    - Order of the parameters is not important
+    - no whitespace around "=" allowed. Whitespace only allowed around the ";" parameter separator
+    - Parameter values are allowed to be quoted
+    '''
+    full_type, *arguments = map(lambda x: x.strip(), mime_string.split(';'))
+    full_type = full_type.lower()
+    try:
+        for i, argument in enumerate(arguments):
+            name, value = argument.split('=')
+            arguments[i] = (
+                name.lower(), value if not quoted_string_pattern.match(value) else value[1:-1]
+            )
+        return (full_type, frozenset(arguments))
+    except ValueError:
+        return (mime_string, frozenset())
```

Comment:
Is this really needed to strip the quoted argument value ?

Reply:
Only the ternary operator on line 107 is needed for that. The rest is needed to accept spaces between the equal signs, to be case insensitive on the mime_string and the parameter name but NOT the parameter value, and to accept the parameters in a random order (thats why I put them in a set, because a set does not have an order).